### ISSUE TITLE
Remarkable ActiveRecord for Rspec 2 / Rails 3

### ISSUE THREAD

**Issue by hosh** [remarkable-rb/remarkable#3]

**Comment by txus** [comment]
Hi hosh, first of all I'm really impressed about your work on this port, which I've been following lately! :) But I just came across an unexpected behavior (maybe I'm just using it wrong way). When I try to validate uniqueness of :email in a Customer model, apparently remarkable_activerecord somehow tries to find the first record with no success (¿). My output here: http://gist.github.com/376316

Thanks!

**Comment by hosh** [comment]
@txusty Ok, I'll check this one out. I think I remember that from the old Remarkable too but never tracked it down then.

**Comment by hosh** [comment]
@txusty Yeah, the uniqueness macro requires you to have an existing object in there. http://github.com/remarkable/remarkable/blob/v4.0.0.alpha1/remarkable_activerecord/lib/remarkable_activerecord/matchers/validate_uniqueness_of_matcher.rb

It is in the docs and that's the way it worked in the old way. That means you do need to create the object in a before(:each) callback somewhere, probably with factories. I don't know of a better way to test for uniqueness off hand, so let me know if you do.

**Comment by txus** [comment]
As I guessed, I hadn't understood and tried to use it without instantiating any object! :) Thanks, I'm going with factories then. Sorry for the inconvenience and congratulations for your work!
